kms/kms.rb

```diff
@@ -0,0 +1,392 @@
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}"
+
+  # Create a CryptoKey for your project keyring
+  new_crypto_key = kms_client.create_project_location_key_ring_crypto_key(resource,
+      Cloudkms::CryptoKey.new(purpose: "ENCRYPT_DECRYPT"), crypto_key_id: crypto_key)
+
+  puts "Create CryptoKey #{crypto_key}"
+  # [END kms_create_cryptokey]
+end
+
+$encrypt = -> (project_id:, key_ring_id:, crypto_key:, location:, input_file:,
+               output_file:) do
+  # [START kms_encrypt]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # crypto_key  = "Name of the cryptoKey"
+  # location    = "The location of the new KeyRing"
+  # input_file  = "File to encrypt"
+  # output_file = "File name to use for encrypted input file"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}"
+
+  # Use the KMS API to encrypt the text
+  response = kms_client.encrypt_crypto_key(resource,
+      Cloudkms::EncryptRequest.new(plaintext: File.read(input_file)))
+
+  # Write the encrypted text to a file
+  File.write output_file, response.ciphertext
+
+  puts "Saved encrypted #{input_file} as #{output_file}"
+  # [END kms_encrypt]
+end
+
+$decrypt = -> (project_id:, key_ring_id:, crypto_key:, location:, input_file:,
+               output_file:) do
+  # [START kms_decrypt]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # crypto_key  = "Name of the cryptoKey"
+  # location    = "The location of the new KeyRing"
+  # input_file  = "The path to an encrypted file"
+  # output_file = "The path to write the decrypted file"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}"
+
+  # Use the KMS API to decrypt the text
+  encrypted_file = File.read(input_file).chomp
```

Comment:
This isn't really an `encrypted_file`, it's `encrypted_text`

I would expect an `encrypted_file` variable to maybe be the path to a file or a File object.

How about `encrypted_text`?